src/am_super/super_node_mediator.cpp

```
@@ -263,4 +268,15 @@ bool SuperNodeMediator::StateTransition::hasLifecycleCommand()
   return life_cycle_command != (LifeCycleCommand)-1;
 }
 
+SuperState SuperNodeMediator::operatorCommandToState(const OperatorCommand& command) 
```

as you know, I don't like switch statements.  This begs for unhandled problems when new commands are added and this switch statement is overlooked and someone ends up with a null value instead of a valid command. 

In fact, you should not return null since that results in undefined behavior.  Instead, you should throw an exception that will be reported properly to a developer that they did something wrong.  You should provide a proactive questions "isSomethingValid" which allows a developer to proactively decide if they will call this method.  If they call it directly with a bad value then the system halts because it is a coding error.

Better yet, you should enforce this through an object oriented relationship rather than a functional programming using  a switch statement so this switch statement should go away in favor of providing this to the Transition object.

Furthermore, you created an untested method. If you want my feedback prior to testing then I appreciate a proactive comment here.  I expect your code to be fully tested and this has no test.

---

I'd consider passing the OperatorCommand to StateTransition associated to the State already managed by StateTransition.  

Provide your "hasOperatorCommand" method on the StateTransition struct. 

```
 struct StateTransition
  {
    StateTransition(SuperState _to_state, std::function<bool(SuperNodeMediator::Supervisor&,SuperNodeMediator::SuperNodeInfo&, SuperNodeMediator&)> _check,
                    LifeCycleCommand _life_cycle_command = (LifeCycleCommand)-1)
    {
      to_state = _to_state;
      check = _check;
      life_cycle_command = _life_cycle_command;
      on_check_result = true; //TODO: remove this; we are assuming the check method should always return true now
    }
```

We can discuss more about this approach if it isn't obvious to you.